src/cmds/fs/sort.c

```diff
@@ -32,33 +32,48 @@ static void print_usage(void) {
 	);
 }
 
-static int compare(const void *fst_ptr, const void *snd_ptr) {
+/* 	This function similar to "strcasecmp" except for "tolower" changed to "toupper",
+	because "sort -f" folds lower case to upper case characters */
+static int case_compare(const char *str1, const char *str2) {
```

Comment:
I figured out that `strcasecmp` still can't do the job well with `-f` option, when there're characters between 'Z' and 'a' (like '[', '\\', ']' and so on). GNU `sort -f` treats `'z' < '['` and `'A' < '['`, while for `strcasecmp` `'[' < 'z'`  and `'[' < 'A'`. It happens because `strcasecmp` folds chars to lower case, while GNU sort folds chars to upper case. So I've rewrited it with just changing `tolower` to `toupper`.